ngraph/core/src/op/squeeze.cpp

```diff
@@ -46,10 +46,31 @@ void op::Squeeze::pre_validate_and_infer_types()
                                       ? axes_constant->cast_vector<int64_t>().empty()
                                       : false;
 
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_element_type(1).is_integral_number(),
+        "Second input (axes) is expected to be of integer type. Got: ",
+        get_input_element_type(1));
+
+    NODE_VALIDATION_CHECK(
+        this,
+        axes_pshape.rank().is_static() && axes_pshape.rank().get_length() < 2,
```

Comment:
```suggestion
        axes_pshape.rank().is_dynamic || axes_pshape.rank().get_length() < 2,
```
or
```suggestion
        axes_pshape.rank().compatible(0) || axes_pshape.rank().compatible(1),
```
Please add type_prop test also to check on dynamic shape for second input